modules/ui/fields/roadspeed.js

```diff
@@ -119,16 +125,20 @@ export function uiFieldRoadspeed(field, context) {
     roadspeed.tags = function(tags) {
         _tags = tags;
 
-        var value = tags[field.key];
+        var rawValue = tags[field.key];
+        var value = rawValue;
         var isMixed = Array.isArray(value);
 
         if (!isMixed) {
-            if (value && value.indexOf('mph') >= 0) {
-                value = parseInt(value, 10).toString();
+            if (rawValue && rawValue.indexOf('mph') >= 0) {
                 _isImperial = true;
-            } else if (value) {
+            } else if (rawValue) {
                 _isImperial = false;
             }
+
+            value = parseInt(value, 10);
+            if (isNaN(value)) value = rawValue;
+            value = formatFloat(value);
```

Comment:
When the tag is not present, this crashes in the floatFormatter as the `number` is `undefined`.

Something like would fixe it:
```suggestion
            if (isNaN(value)) {
                value = rawValue;
            } else {
                value = formatFloat(value);
            }
```